fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerUniqueIndexTest.java

```diff
@@ -366,4 +371,184 @@ public void resolveUniquenessViolations() {
             context.commit();
         }
     }
+
+    @Test
+    public void testReadableUniquePendingSingle() {
```

Comment:
There were a bunch of changes to `FDBRecordStore`, It would probably be good to add some tests of those methods directly (which would hopefully expose the fact that `markIndexReadableOrUniquePending` returns `true` even if the index was already `READABLE_UNIQUE_PENDING`.
I think you should be able to avoid using the indexer by:
1. creating a unique index
2. saving some valid records
3. marking the index write only
4. saving an invalid record
5. trying to mark the index readable, or `READABLE_UNIQUE_PENDING`

It would probably be good to test the addition of uniqueness violations on an index in either of those states, and that you can get back the right set of uniqueness violations. 
If there isn't already a test in WriteOnly state, it would probably be good to test adding and removing multiple violations on the same key in both of these states.

Reply:
It would probably be good to have tests that add more uniquness violations, both by adding records that have the same violations as existing records, and adding completely new violations. Resolving all should be required before updating to readable.

Reply:
Another thing to test on the store directly would be scanning the index when it has uniqueness violations, and seeing those violations.
It would probably also be good to test this for all index types, particularly rank and bitmaps support the unique option, but bitmaps, I don't think support `READABLE_UNIQUE_PENDING` (which is fine, I don't think we need to add support for that here)

Reply:
It would probably also be good to add tests that try to do `markIndexRadableOrUniquePending` when the index is only partially built, but has no uniqueness violations.

Reply:
Most applied - but scanning for uniqueness violations. 
Do you mean index scan (or scrub)? It wouldn't discover UV, wouldn't it?